src/hooks/useWalletConnect.ts

```diff
@@ -136,6 +148,16 @@ export default function useWalletConnect() {
 
   const onSessionProposal = useCallback(
     (proposal: Web3WalletTypes.SessionProposal) => {
+      void web3wallet.core.pairing.updateMetadata({
+        topic: proposal.params.pairingTopic,
+        metadata: {
+          ...proposal.params.proposer.metadata,
+          verifyUrl: proposal.verifyContext.verified.isScam
+            ? "SCAM"
+            : proposal.verifyContext.verified.validation,
+        },
+      });
```

Comment:
It's a bit strange that we have to do that instead of just using their data but we can take a look at this later

Reply:
We could add a TODO comment to not forget this